tests/integ/test_dataframe.py

```diff
@@ -1583,3 +1583,18 @@ def test_unpivot(session, column_list):
         ],
         sort=False,
     )
+
+
+def test_create_dataframe_string_length(session):
+    table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+    df = session.create_dataframe(["ab", "abc", "abcd"], schema=["a"])
+    df.write.mode("overwrite").save_as_table(table_name, create_temp_table=True)
+    datatype = json.loads(
+        session.sql(f"show columns in {table_name}").collect()[0]["data_type"]
+    )
+    assert datatype["type"] == "TEXT"
+    assert datatype["length"] == 2**20 * 16  # max length (16 MB)
+    session.sql(f"insert into {table_name} values('abcde')").collect()
```

Comment:
Unrelated to this PR: I found we have `update` and `delete` method for table https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/_autosummary/snowflake.snowpark.html#snowflake.snowpark.Table, but we don't have `insert` method (we have `insert` when merging two tables), so I have to use SQL here. Do you think we should add it? Note that scala snowpark and databricks' delta table doesn't have it. https://docs.databricks.com/delta/delta-update.html#language-python